community/kernel/src/main/java/org/neo4j/diagnostics/DiagnosticsReporter.java

```diff
@@ -75,7 +75,23 @@ public void dump( Set<String> classifiers, Path destination, DiagnosticsReporter
         }
 
         // Make sure target directory exists
-        Files.createDirectories( destination.getParent() );
+        Path destinationFolder = destination.getParent();
+        Files.createDirectories( destinationFolder );
+
+        // Estimate an upper bound of the final size and make sure it will fit
+        long estimatedFinalSize = sources.stream().mapToLong(
+                diagnosticsReportSource -> diagnosticsReportSource.estimatedSize( progress ) ).sum();
+        long freeSpace = destinationFolder.toFile().getFreeSpace();
+        if ( estimatedFinalSize > freeSpace )
+        {
+            String message =
+                    String.format( "WARNING: Free available disk space for %s is %s, worst case estimate is %s",
+                            destination.getFileName(), Format.bytes( freeSpace ), Format.bytes( estimatedFinalSize ) );
+            if ( !progress.shouldIgnorePotentialFullDisk( message ) )
```

Comment:
let's change this condition from `!shouldIgnore` to `continueOnFull` to remove negation

Reply:
Still a negation though :sweat_smile: 
Choosing `stopOnPotentialFullDisk`